src/public/js/web/gmaps.js

```diff
@@ -0,0 +1,2343 @@
+                    scaleControl: scaleControl,
+                    streetViewControl: streetViewControl,
+                    overviewMapControl: overviewMapControl,
+                };
+
+            if (
+                typeof options.el === "string" ||
+                typeof options.div === "string"
+            ) {
+                this.el = getElementById(container_id, options.context);
+            } else {
+                this.el = container_id;
+            }
+
+            if (typeof this.el === "undefined" || this.el === null) {
+                throw "No element defined.";
+            }
+
+            window.context_menu = window.context_menu || {};
+            window.context_menu[self.el.id] = {};
+
+            this.controls = [];
+            this.overlays = [];
+            this.layers = []; // array with kml/georss and fusiontables layers, can be as many
+            this.singleLayers = {}; // object with the other layers, only one per layer
+            this.markers = [];
+            this.polylines = [];
+            this.routes = [];
+            this.polygons = [];
+            this.infoWindow = null;
+            this.overlay_el = null;
+            this.zoom = options.zoom;
+            this.registered_events = {};
+
+            this.el.style.width =
+                options.width || this.el.scrollWidth || this.el.offsetWidth;
+            this.el.style.height =
+                options.height || this.el.scrollHeight || this.el.offsetHeight;
+
+            google.maps.visualRefresh = options.enableNewStyle;
+
+            for (i = 0; i < options_to_be_deleted.length; i++) {
+                delete options[options_to_be_deleted[i]];
+            }
+
+            if (options.disableDefaultUI !== true) {
+                map_base_options = extend_object(
+                    map_base_options,
+                    map_controls_options
+                );
+            }
+
+            map_options = extend_object(map_base_options, options);
+
+            for (i = 0; i < events_that_hide_context_menu.length; i++) {
+                delete map_options[events_that_hide_context_menu[i]];
+            }
+
+            for (i = 0; i < events_that_doesnt_hide_context_menu.length; i++) {
+                delete map_options[events_that_doesnt_hide_context_menu[i]];
+            }
+
+            this.map = new google.maps.Map(this.el, map_options);
+
+            if (markerClustererFunction) {
+                this.markerClusterer = markerClustererFunction.apply(this, [
+                    this.map,
+                ]);
+            }
+
+            var buildContextMenuHTML = function (control, e) {
+                var html = "",
+                    options = window.context_menu[self.el.id][control];
+
+                for (var i in options) {
+                    if (options.hasOwnProperty(i)) {
+                        var option = options[i];
+
+                        html +=
+                            '<li><a id="' +
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/rikai-academy/Foods_Drinks_01/pullRequest?prid=7829589)
```suggestion
                            "<li><a id=\"" +
```